wordpress-mobile/WordPress-Android
# Issue/13329 backup handlers

Parent #13329 

This PR adds:
- `BackupDownloadHandler` to handle the backup download request via `PostBackupDownloadUseCase`.
- `BackupDownloadStatusHandler` to  handle the backup download request via `BackupDownloadStatusUseCase `.
The above introduces `flow` and `emit` to handle progress updates.
- Starts to build out the wizard `BackupDownloadState` which gets passed to Progress & Complete views.
- Starts to build out `BackupDownloadProgressViewModel` and `BackupDownloadProgressFragment`
 
**Merge Notes**
- There is an existing[ FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1821) that needs to be merged into develop first.  

**Note:**
- There is no view in Progress or Complete. The snackbar is used to show progress for the time being. My next task will build out the progress pieces.

**To Test**
(To test locally before the FluxC PR is merged, pull down the [branch](https://github.com/wordpress-mobile/WordPress-FluxC-Android/tree/issue/13329-adjust-backup-download-gseon-type) or point to FluxC Commit 7590938)
- Launch the app with BackupDownloadConfig flag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Download Backup menu item
- Select items to backup and then tap the Create Downloadable File Button
- Note that the progress view slides in and the snackbar is updated with progress. When download is ready, the view will slide out to complete.

PR submission checklist:
